api/v1alpha3/machine_types.go

```diff
@@ -37,6 +37,18 @@ const (
 
 	// MachineDeploymentLabelName is the label set on machines if they're controlled by MachineDeployment
 	MachineDeploymentLabelName = "cluster.x-k8s.io/deployment-name"
+
+	// PreDrainDeleteHookAnnotationPrefix annotation specifies the prefix we
+	// search each annotation for during the pre-drain.delete lifecycle hook
+	// to pause reconciliation of deletion. These hooks will prevent removal of
+	// draining the associated node until all are removed.
+	PreDrainDeleteHookAnnotationPrefix = "pre-drain.delete.hook.machine.cluster.x-k8s.io"
```

Comment:
```suggestion
	PreDrainDeleteHookAnnotationPrefix = "pre-drain-delete-hook.machines.cluster.x-k8s.io"
```
?

Reply:
This suggestion would be contrary to the CAEP.

Reply:
Can we update both? Having a string with no subdomain makes parsing it way easier.

Reply:
Bump - what do we want to do here? @michaelgugino @vincepri @CecileRobertMichon @rudoi @detiber @fabriziopandini.

Reply:
Either way is fine I guess, I suggested a plain string without dots because I wanted to avoid to suggest that other categories of hooks are going to be supported this way, but if you're fine with the current code, let's just merge it.

Reply:
I think I'd vote in favor of merging as is. We can revisit the naming in v1alpha4 or later.

Reply:
I prefer the dots.  Indeed, I felt that hooks generally should be applied in a hierarchical manner.  pre-drain step of the delete step, is a hook.  There might be pre-drain.reboot if we add power states.  Similarly, there is pre-terminate.delete which indicates it's also a part of the 'delete' portion of the lifecycle.

Maybe the 'pre' portion is redundant?  Though, if we were to add 'post-drain' via some finer-level status indication, then pre would not be redundant.  Same with post-terminate.  Might want to do something after an instance is deleted (what, I have no idea).

Naturally, I propose merging as-is.

Reply:
As-is works for me.

Reply:
+1 from me to as-is